Source/Common/gdcmByteSwap.txx

```diff
@@ -106,7 +106,7 @@ void Swap4(T &a, SwapCode const &swapcode)
 {
 #ifndef GDCM_WORDS_BIGENDIAN
   if ( swapcode == 4321 || swapcode == 2143 )
-    a = (T)(( a << 8 ) | ( a >> 8 ));
+    a = ( a << 8 ) | ( a >> 8 );
```

Comment:
@malaterre I see you didn't take this change. It was meant to make line 109 the same as line 112. I don't see why these two lines should be different...